src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -331,10 +352,22 @@ private static string NormalizeGuidCollectionItems(string bracketLiteralText)
         private static string NormalizeDateTimeCollectionItems(string bracketLiteralText)
         {
             string normalizedText = bracketLiteralText.Substring(1, bracketLiteralText.Length - 2).Trim();
+
+            // If we have empty brackets ()
             if (normalizedText.Length == 0)
             {
                 return "[]";
             }
+
+            string normalizedTextWithoutSpaces = normalizedText.Replace(" ", string.Empty);
+
+            // If we have '' or "" or empty strings in the brackets e.g (' ')
+            // We throw an exception since empty strings are not valid guid values.
+            if (normalizedTextWithoutSpaces == "''" || normalizedTextWithoutSpaces == "\"\"")

```

Comment:
Is there a general way for checking for invalid values? Why do we have special treatment for empty strings? I assume there are other non-empty string values which are not valid GUID or dates. Why don't we do the validation at the same place?

Reply:
+1, lines 362 to 370 should be factored in a helper method since this seems to be useful for other NormalizeXYZcollectionItems.

Reply:
@habbes Yes there is a general way to check for invalid values. I have added tests that confirms that